desci-server/src/utils/driveUtils.ts

```diff
@@ -207,12 +222,32 @@ export async function getTreeAndFill(
       };
       cidInfoMap[ref.cid] = entryDetails;
     });
-    const promises = pubEntries.map(async (ref) => {
+
+    // blockTimeCache prevents redoing the same ceramic node/redis lookup for every pubRef entry if they share the same tx/commit hash
+    // getBlockTime returning null is valid, and its a valid cache value in the blockTimeCache map, to prevent spamming failing lookups in the ceramic node
+    const blockTimeCache = {};
+    const uniqueTxOrCommits = new Set(
+      pubEntries.map((entry) => entry.nodeVersion.transactionId ?? entry.nodeVersion.commitId).filter(Boolean),
+    );
```

Comment:
some NV's have both, since we did double publish for a while. But in case of a backfill migration, the tx ID is probably more correct as it's older